### DISCUSSION TITLE
Handling of overlay scrollbar in editors and views

### DISCUSSION THREAD

**Discussion by ptziegler** [eclipse-platform/eclipse.platform.ui#2635] [General]
First of all, this question is only relevant for Linux, as neither Windows nor MacOS support overlay scrolling.

GTK provides a feature where instead of having a dedicated scrollbar, it is instead shown on top of the client area. From my perspective, this is horrible in terms of usability, as it makes it extremely difficult to click on e.g. the last element in a list, because it is obscured by the horizontal scrollbar.

![image](https://github.com/user-attachments/assets/99617f7b-771a-4117-87f1-f8ac67d7b941)

At least on KDE, this feature has been disabled in several application such as the Dolphin file browser, with the Eclipse IDE being one of the odd ones standing out.

![image](https://github.com/user-attachments/assets/9d974d3b-90f9-4871-8274-dd90e7a68296)

As far as I know, there are two ways this can be disabled. Once by setting the `GTK_OVERLAY_SCROLLING=0` system property which disables overlay scrolling for **all** windows, or by calling `Scrollable.setScrollbarsMode(SWT.NONE)` when creating an SWT composite.

My question: Am I the only one who thinks that this leads to a poor user experience or are there others who actually like the way this works? Would it make sense to change it or to at least make it configurable?

**Comment by laeubi** [comment]
> My question: Am I the only one who thinks that this leads to a poor user experience or are there others who actually like the way this works? Would it make sense to change it or to at least make it configurable?

We can at least start with a global `Scrollable.setDefaultScrollbarsMode(...)` in SWT ... beside that under Debian/Gnome this seems already be the case to not use overlay scrollbar so I can't judge how bad it is.

**Comment by iloveeclipse** [comment]
It is horrible but it is coming from GTK devs that know better about usability (sarcasm).
I believe they look at Windows and MacOS and implement random cool UI features from these in a most annoying way in GTK.

We disable that "feature" by default in our IDE's and the products based on Eclipse.
